Error compiling interval code

On (Gentoo) Linux with GCC 12.3.1 and boost-1.81.0 compiling interval-enabled ESBMC on current master stops with this error:
```
/home/kane/scorch/esbmc/src/goto-programs/abstract-interpretation/interval_domain.cpp: In member function 'Interval interval_domaint::get_interval_from_const(const expr2tc&) [with Interval = interval_templatet<boost::multiprecision::number<boost::multiprecision::backends::cpp_bin_float<100> > >; expr2tc = irep_container<expr2t>]':
/home/kane/scorch/esbmc/src/goto-programs/abstract-interpretation/interval_domain.cpp:84:7: error: 'isnormal' was not declared in this scope
   84 |   if(!isnormal(real_value.to_double()) || real_value.is_zero())
      |       ^~~~~~~~
/home/kane/scorch/esbmc/src/goto-programs/abstract-interpretation/interval_domain.cpp:84:7: note: suggested alternatives:
In file included from /usr/include/boost/cstdfloat.hpp:27,
                 from /usr/include/boost/multiprecision/detail/float128_functions.hpp:15,
                 from /usr/include/boost/multiprecision/detail/fpclassify.hpp:13,
                 from /usr/include/boost/multiprecision/detail/default_ops.hpp:15,
                 from /usr/include/boost/multiprecision/detail/generic_interconvert.hpp:12,
                 from /usr/include/boost/multiprecision/number.hpp:12,
                 from /usr/include/boost/multiprecision/cpp_int.hpp:19,
                 from /usr/include/boost/multiprecision/cpp_bin_float.hpp:15,
                 from /home/kane/scorch/esbmc/src/goto-programs/abstract-interpretation/interval_domain.h:14,
                 from /home/kane/scorch/esbmc/src/goto-programs/abstract-interpretation/interval_domain.cpp:4:
/usr/include/boost/math/cstdfloat/cstdfloat_cmath.hpp:728:62: note:   'std::isnormal'
  728 |             inline bool                                      isnormal   BOOST_PREVENT_MACRO_SUBSTITUTION(boost::math::cstdfloat::detail::float_internal128_t x) { return boost::math::cstdfloat::detail::fpclassify BOOST_PREVENT_MACRO_SUBSTITUTION(x) == FP_NORMAL; }
      |                                                              ^~~~~~~~
In file included from /home/kane/scorch/esbmc/debug/_deps/fmt-src/include/fmt/format.h:38,
                 from /home/kane/scorch/esbmc/src/util/message.h:14,
                 from /home/kane/scorch/esbmc/src/irep2/irep2_utils.h:8,
                 from /home/kane/scorch/esbmc/src/goto-programs/goto_program.h:9,
                 from /home/kane/scorch/esbmc/src/goto-programs/abstract-interpretation/ai_domain.h:7,
                 from /home/kane/scorch/esbmc/src/goto-programs/abstract-interpretation/ai.h:10,
                 from /home/kane/scorch/esbmc/src/goto-programs/abstract-interpretation/interval_domain.h:7:
/usr/lib/gcc/x86_64-pc-linux-gnu/12/include/g++-v12/cmath:638:3: note:   'std::isnormal'
  638 |   isnormal(float __x)
      |   ^~~~~~~~
/usr/include/boost/multiprecision/detail/default_ops.hpp:2199:38: note:   'boost::multiprecision::isnormal'
 2199 | inline BOOST_MP_CXX14_CONSTEXPR bool isnormal BOOST_PREVENT_MACRO_SUBSTITUTION(const multiprecision::detail::expression<tag, A1, A2, A3, A4>& arg)
      |                                      ^~~~~~~~
/usr/include/boost/multiprecision/detail/default_ops.hpp:2194:38: note:   'boost::math::isnormal'
 2194 | inline BOOST_MP_CXX14_CONSTEXPR bool isnormal BOOST_PREVENT_MACRO_SUBSTITUTION(const multiprecision::number<Backend, ExpressionTemplates>& arg)
      |                                      ^~~~~~~~
/usr/include/boost/math/cstdfloat/cstdfloat_cmath.hpp:728:62: note:   'boost::math::cstdfloat::detail::isnormal'
  728 |             inline bool                                      isnormal   BOOST_PREVENT_MACRO_SUBSTITUTION(boost::math::cstdfloat::detail::float_internal128_t x) { return boost::math::cstdfloat::detail::fpclassify BOOST_PREVENT_MACRO_SUBSTITUTION(x) == FP_NORMAL; }
      |                                                              ^~~~~~~~
```
This patch solves the problem for me:
```diff
diff --git a/src/goto-programs/abstract-interpretation/interval_domain.cpp b/src/goto-programs/abstract-interpretation/interval_domain.cpp
index e5fb8c663a..e947997680 100644
--- a/src/goto-programs/abstract-interpretation/interval_domain.cpp
+++ b/src/goto-programs/abstract-interpretation/interval_domain.cpp
@@ -1,6 +1,9 @@
 /// \file
 /// Interval Domain
 // TODO: Ternary operators, lessthan into lessthanequal for integers
+
+#include <cmath>
+
 #include <goto-programs/abstract-interpretation/interval_domain.h>
 #include <util/arith_tools.h>
 #include <util/c_typecast.h>
@@ -69,8 +72,6 @@ integer_intervalt interval_domaint::get_interval_from_const(const expr2tc &e)
   return result;
 }
 
-#include <cmath>
-
 template <>
 real_intervalt interval_domaint::get_interval_from_const(const expr2tc &e)
 {
@@ -81,7 +82,7 @@ real_intervalt interval_domaint::get_interval_from_const(const expr2tc &e)
   auto real_value = to_constant_floatbv2t(e).value;
 
   // Health check, is the convertion to double ok? See #1037
-  if(!isnormal(real_value.to_double()) || real_value.is_zero())
+  if(!std::isnormal(real_value.to_double()) || real_value.is_zero())
   {
     if(real_value.is_double())
       log_warning("ESBMC fails to to convert {} into double", *e);
```
@rafaelsamenezes As you are working on the interval code right now, could I ask you to integrate this or a similar patch?

--- a/src/goto-programs/abstract-interpretation/interval_domain.cpp
+++ b/src/goto-programs/abstract-interpretation/interval_domain.cpp
@@ -70,8 +70,6 @@ interval_domaint::get_interval_from_const(const expr2tc &e) const
   return result;
 }
 
-#include <cmath>
-
 template <>
 real_intervalt interval_domaint::get_interval_from_const(const expr2tc &e) const
 {
@@ -82,7 +80,7 @@ real_intervalt interval_domaint::get_interval_from_const(const expr2tc &e) const
   auto real_value = to_constant_floatbv2t(e).value;
 
   // Health check, is the convertion to double ok? See #1037
-  if(!isnormal(real_value.to_double()) || real_value.is_zero())
+  if(!std::isnormal(real_value.to_double()) || real_value.is_zero())
   {
     if(real_value.is_double())
       log_warning("ESBMC fails to to convert {} into double", *e);

--- a/src/goto-programs/abstract-interpretation/interval_domain.cpp
+++ b/src/goto-programs/abstract-interpretation/interval_domain.cpp
@@ -70,6 +70,8 @@ interval_domaint::get_interval_from_const(const expr2tc &e) const
   return result;
 }
 
+#include <cmath>
+
 template <>
 real_intervalt interval_domaint::get_interval_from_const(const expr2tc &e) const
 {

